### ISSUE TITLE
scala-steward detects that it should update a version but then fails to bump it

### ISSUE THREAD

**Issue by alexklibisz** [scala-steward-org/scala-steward#2836]
I'm working on setting up scala-steward for some internal projects. There is one in particular that is showing some very strange behavior:

The version is defined as a single `private val FooVersion = "22.12.2"` in a `project/Dependencies.scala` file. This val is then re-used to define nine different dependencies within the same file.

Scala steward recognizes that it should update the version:

```
2022-12-17 00:53:01,826 INFO  Found 9 updates:
com.foo:(aaa, aaa_2.13) : 22.12.2 -> 22.12.4
com.foo:(bbb, bbb_2.13) : 22.12.2 -> 22.12.4
com.foo:(ccc, ccc_2.13) : 22.12.2 -> 22.12.4
com.foo:(ddd, ddd_2.13) : 22.12.2 -> 22.12.4
com.foo:(eee, eee_2.12) : 22.12.2 -> 22.12.4
com.foo:(fff, fff_2.13) : 22.12.2 -> 22.12.4
com.foo:(ggg, ggg_2.13) : 22.12.2 -> 22.12.4
com.foo:(hhh, hhh_2.13) : 22.12.2 -> 22.12.4
com.foo:(iii, iii_2.13) : 22.12.2 -> 22.12.4
```

But then it fails to do so, all nine times:

```
2022-12-17 00:53:04,617 INFO  Trying heuristic 'moduleId'
2022-12-17 00:53:04,628 INFO  Trying heuristic 'strict'
2022-12-17 00:53:04,632 INFO  Trying heuristic 'original'
2022-12-17 00:53:04,637 INFO  Trying heuristic 'relaxed'
2022-12-17 00:53:04,642 INFO  Trying heuristic 'sliding'
2022-12-17 00:53:04,648 INFO  Trying heuristic 'completeGroupId'
2022-12-17 00:53:04,653 INFO  Trying heuristic 'groupId'
2022-12-17 00:53:04,713 INFO  Trying heuristic 'specific'
2022-12-17 00:53:04,714 WARN  Unable to bump version
```

(The above is repeated nine times exactly)

So it seems like the UpdateHeuristics are failing to replace the version. But I'm very confused how that's possible given it was able to find the versions in the first place. It should only have to update that single `private val`.

This has happened on 0.16.1 and 0.17.0. I haven't tried others.

LMK if there's a suggested way to diagnose and/or reproduce. Obviously I'm not going to post the full project, but maybe I can provide some approximation of it. But I'm not exactly sure what's the best way to package that up.

**Comment by alexklibisz** [comment]
Maybe this is fixed as part of https://github.com/scala-steward-org/scala-steward/pull/2816? That would be great timing. Is there any way for me to verify?

**Comment by fthomas** [comment]
> Is there any way for me to verify?

You could check out that branch and run Scala Steward from there. You could also add a test to `RewriteTest` on that branch and check if it passes. But from what you are describing, I'm confident that it is fixed  by #2816.

**Comment by alexklibisz** [comment]
Confirming that #2816 fixed the issue. Thanks!
